packages/@aws-cdk/aws-stepfunctions-tasks/lib/sns/publish.ts

```diff
@@ -77,6 +133,17 @@ export class SnsPublish extends sfn.TaskStateBase {
       }
     }
 
+    if (props.messageAttributes) {
+      const attributes: { [key: string]: any } = {};
+      for (const attrib of props.messageAttributes) {
+        const dataType = attrib.type || SnsMessageAttributeType.STRING;
```

Comment:
prefer ?? for nullish coalescing, but see other comments since this will probably change
```suggestion
        const dataType = attrib.type ?? SnsMessageAttributeType.STRING;
```